### ISSUE TITLE
[OS X] "Server is temporarily unavailable" forever (client doesn't seem to check server health)

### ISSUE THREAD

**Issue by tessus** [owncloud/client#4561]
### Expected behaviour
After the server comes out of maintenance mode (or becomes available again, generally speaking), the client should start syncing again.

### Actual behaviour
The client shows `Server xxx is temporarily unavailable` and does nothing.

### Steps to reproduce
1. put server in maintenance mode
2. wait 10 minutes
3. bring server out of maintenance mode

### Server configuration
Operating system: Linux 64bit

Web server: Apache 2.4.18

Database: Mysql 5.6.25

PHP version: 5.6.19

ownCloud version: 8.2.3

Storage backend: file system

### Client configuration
Client version: Version 2.1.1 (build 3107)

Operating system: osx 10.9.5 and 10.11.3

OS language: us-en

Installation path of client: /Applications/owncloud.app

### Logs

1. Output of `owncloud --logwindow` or `owncloud --logfile log.txt`:
I can't get a log file, because as soon as I start the client with above commands a new instance of the client is started and it starts syncing. The other client is still running, but also still in the state of `"Server is temporarily unavailable`

2. Web server error log: no errors

3. ownCloud log (data/owncloud.log): no errors 

### Comments

It seems to me that the client does not check periodically, whether the server is available.
I have to close the client app and start it again to function properly. I think this also happened, when I stopped the web server for a while. After starting it again, the client never came back to a normal state.

**Comment by tflidd** [comment]
I have observed similar issues on OS X 10.9.5 and version 2.1.1. It's often when I need to connect to a network which needs some time (authentication on first connection via browser on a wired connection). Mostly, the client recovers and starts syncing again (perhaps after 5-10 minutes), in some cases, I need to restart the client after 15 minutes.

I can try to gather some logfiles next time, if there is anything to get.

OC 8.1.5, no encryption, no external storage.

**Comment by guruz** [comment]
Can you guys try a current nightly?
We have a fix related to this in there..

**Comment by tessus** [comment]
@guruz I can install a nightly version, but I can't test it, since I'd have to actually stop the server for that. I will be able to do that in the next maintenance window, but this isn't scheduled for another 2 months.

**Comment by tflidd** [comment]
@guruz I could. I'm just wondering what's the most efficient way to do this as I can't reproduce this situation. If there is a testpilot-version, I could run a second client in parallel and then see if there is a situation where the current client fails.

**Comment by guruz** [comment]
@tflidd http://download.owncloud.com/desktop/daily/testpilotcloud-2.2.0.3167-nightly20160309.pkg :)

**Comment by tflidd** [comment]
@guruz
I installed the testpilot client. My working user has no admin 
privileges, so for the installation process I needed to authenticate the 
admin user. Installation didn't create any error. After the setup, I try 
to open the client as unprivileged user but the client crashes (crash 
report was sent). If I start testpilotclient from terminal I get this 
error
````
m_crashReporterChar:  
/Applications/testpilotcloud.app/Contents/MacOS/testpilotcloud_crash_reporter
2016-03-11 15:02:17.615 testpilotcloud[4149:507] An uncaught exception 
was raised
2016-03-11 15:02:17.615 testpilotcloud[4149:507] You must specify the 
URL of the appcast as the SUFeedURL key in either the Info.plist or the 
user defaults!
2016-03-11 15:02:17.615 testpilotcloud[4149:507] (
	0   CoreFoundation                      0x00007fff947d825c 
__exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff917c1e75 
objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff947d810c 
+[NSException raise:format:] + 204
	3   Sparkle                             0x00000001042505a4 -[SUUpdater 
feedURL] + 340
	4   Sparkle                             0x000000010424e6e8 -[SUUpdater 
initForBundle:] + 757
	5   Sparkle                             0x000000010424e3d1 +[SUUpdater 
updaterForBundle:] + 205
	6   Sparkle                             0x000000010424e2de +[SUUpdater 
sharedUpdater] + 64
	7   testpilotcloud                      0x00000001014eaf17 
_ZN3OCC14SparkleUpdaterC2ERK7QString + 119
	8   testpilotcloud                      0x00000001014ea0be 
_ZN3OCC7Updater6createEv + 606
	9   testpilotcloud                      0x00000001014e9e55 
_ZN3OCC7Updater8instanceEv + 21
	10  testpilotcloud                      0x0000000101436739 
_ZN3OCC15GeneralSettings14slotUpdateInfoEv + 25
	11  testpilotcloud                      0x0000000101436251 
_ZN3OCC15GeneralSettingsC2EP7QWidget + 449
	12  testpilotcloud                      0x00000001014c9795 
_ZN3OCC17SettingsDialogMacC2EPNS_11ownCloudGuiEP7QWidget + 2293
	13  testpilotcloud                      0x000000010144a768 
_ZN3OCC11ownCloudGuiC2EPNS_11ApplicationE + 88
	14  testpilotcloud                      0x00000001013fa211 
_ZN3OCC11ApplicationC2ERiPPc + 1569
	15  testpilotcloud                      0x00000001013e91a9 main + 57
	16  testpilotcloud                      0x00000001013e8f24 start + 52
	17  ???                                 0x0000000000000001 0x0 + 1
)
2016-03-11 15:02:17.616 testpilotcloud[4149:507] *** Terminating app due 
to uncaught exception 'SUNoFeedURL', reason: 'You must specify the URL 
of the appcast as the SUFeedURL key in either the Info.plist or the user 
defaults!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff947d825c 
__exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff917c1e75 
objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff947d810c 
+[NSException raise:format:] + 204
	3   Sparkle                             0x00000001042505a4 -[SUUpdater 
feedURL] + 340
	4   Sparkle                             0x000000010424e6e8 -[SUUpdater 
initForBundle:] + 757
	5   Sparkle                             0x000000010424e3d1 +[SUUpdater 
updaterForBundle:] + 205
	6   Sparkle                             0x000000010424e2de +[SUUpdater 
sharedUpdater] + 64
	7   testpilotcloud                      0x00000001014eaf17 
_ZN3OCC14SparkleUpdaterC2ERK7QString + 119
	8   testpilotcloud                      0x00000001014ea0be 
_ZN3OCC7Updater6createEv + 606
	9   testpilotcloud                      0x00000001014e9e55 
_ZN3OCC7Updater8instanceEv + 21
	10  testpilotcloud                      0x0000000101436739 
_ZN3OCC15GeneralSettings14slotUpdateInfoEv + 25
	11  testpilotcloud                      0x0000000101436251 
_ZN3OCC15GeneralSettingsC2EP7QWidget + 449
	12  testpilotcloud                      0x00000001014c9795 
_ZN3OCC17SettingsDialogMacC2EPNS_11ownCloudGuiEP7QWidget + 2293
	13  testpilotcloud                      0x000000010144a768 
_ZN3OCC11ownCloudGuiC2EPNS_11ApplicationE + 88
	14  testpilotcloud                      0x00000001013fa211 
_ZN3OCC11ApplicationC2ERiPPc + 1569
	15  testpilotcloud                      0x00000001013e91a9 main + 57
	16  testpilotcloud                      0x00000001013e8f24 start + 52
	17  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
m_crashReporterWChar: 
/Applications/testpilotcloud.app/Contents/MacOS/testpilotcloud_crash_reporterDump 
file was written to: 
/var/folders/mj/j_2czjsx261_yg6zz331_s2h0000gp/T/CD5D8A7F-F95F-442F-8F57-E1FA89076D3D.dmp
````

**Comment by guruz** [comment]
Installing the testpilot works fine for me (as admin user).
Could you paste the output of:
```find /Applications/testpilotcloud.app```
?

**Comment by tflidd** [comment]
Here you go: http://pastebin.com/GtviVYyC
In the meantime, I logged in as the admin-user, client crashes as well, 
report was sent.

I'm still on OS X 10.9.5 if that matters.

**Comment by guruz** [comment]
That crash might be related to @ckamm and 17003cec198884c4072e2411226ff52de58aa16e .. however before @ckamm 's patch it should crash too. Strange. Checking.

Need to talk to @danimo about Sparkle.

**Comment by guruz** [comment]
@danimo says it could also be the Sparkle update (https://github.com/owncloud/client/issues/2965#issuecomment-195448501)

Because that updater URL should always be specified (and it is even for the testpilotcloud..) so it should not crash..
Maybe it's really the OS X version somehow to be blamed.

**Comment by tessus** [comment]
@guruz the osx version might be the reason for the crash (Sparkle's fault or not), but the original problem also occurs on 10.11.3 (as stated in my problem dewsciption).

**Comment by guruz** [comment]
@tessus That's clear :) 

@jturcotte ^^ Do you want to try to reproduce the crash?

**Comment by jturcotte** [comment]
@guruz I also get the crash, I can't verify which version of Sparkle I have in the package, but this could seems to be https://github.com/sparkle-project/Sparkle/issues/767 which should be present in 1.13 and fixed in 1.14.

**Comment by guruz** [comment]
OK, we're going to update Sparkle and then give @tflidd a new build to test for the crash.

And meanwhole we wait for @tessus on the reconnection issue.

**Comment by guruz** [comment]
@tflidd We have updated Sparkle.. could you try if this still crashes for you? 
http://download.owncloud.com/desktop/daily/testpilotcloud-2.2.0.3246-nightly20160406.pkg

If it does not crash, could you then also test the original network issue that @tessus had reported?

thank you

**Comment by tflidd** [comment]
great. I will test it. Unfortunately (luckily), I didn't have any problems with my 2.1.1-client since I reported the problem here. Perhaps my network has improved. I will try the new testpilot in order to reproduce the problems with the maintenance mode this weekend.

**Comment by guruz** [comment]
@tflidd thanks. Also for testing if your [testpilot crash](https://github.com/owncloud/client/issues/4561#issuecomment-195378992) is fixed, not only the network reconnect issue

**Comment by tflidd** [comment]
I installed the testpilot client along with my normal client (2.1.1). Both sync as they should, I can put the server in maintenance mode (OC 9.0.1) for several minutes, disable my network connection, it takes a couple of seconds to reconnect (testpilot sees the changes about 10s faster than the 2.1.1-client).

I'm not able to reproduce a situation where the 2.1.1-client stalls.

**Comment by tessus** [comment]
Please note that @tflidd's scenario is different from the one I created the ticket for. When I run @tflidd's scenario everything even works fine with the 2.1.1 client for me.

The important part is this:
-     **put server in maintenance mode**
-     wait 10 minutes
-     bring server out of maintenance mode

In this case, the client never reconnects. It just says "Server is temporarily unavailable".

**Comment by guruz** [comment]
@tessus But @tflidd writes " I can put the server in maintenance mode, ...". I guess he meant putting server in maintenance mode OR disabling network connection?

**Comment by tflidd** [comment]
It's true, my scenario was different from yours (sorry for hijacking your topic, I thought it might be related) @tessus. But I tried to reproduce your situation with the maintenance mode as well. I couldn't manage to reproduce this scenario. I tried the MacBook to be active all the time, I tried to get it into screensaver-mode, ... Can you give more details? Do you use a wifi connection? Is it the same if you interrupt your network connection? Is it a public server or something with a local ip? No proxy, no VPN?

You could use tcpdump on your server in order to see what your client is doing.

**Comment by guruz** [comment]
@tessus And you are saying a recent nightly does still not fix it for you?

**Comment by tessus** [comment]
@guruz why are we talking about 3 issues in this ticket? can we move @tflidd's 2 issues to a new ticket and get back to the original problem?

I will have a maintenance window in about six hours, thus I can test the nightly then.
The link above does not work, but I found a newer one: http://download.owncloud.com/desktop/daily/ownCloud-2.2.0.3249-nightly20160408.pkg 
Shall I use that one for testing?

**Comment by tessus** [comment]
@tflidd I only see this problem, when the server is put in maintenance mode for a while (about 10 minutes). 
Anything client related (interrupted network, screensaver, VPN, no-VPN) does not break the client. After a few seconds I have a connection again.

**Comment by tessus** [comment]
@guruz tested with _Version 2.2.0-nightly20160408 (build 3249)_ and the problem seems to be fixed on OSX 10.11.x. I haven't been able to test with 10.9.x. I think we can close this for now. Cheers.

**Comment by dragotin** [comment]
@rperezb can qa help us verifying this ultimately? Does the client version pre 2.2.0 properly come back after the server was in 503 *Maintenance* mode? Thanks

**Comment by tessus** [comment]
@guruz found another problem with the nightly build though.

**Comment by rperezb** [comment]
@dragotin tomorrow Miguel joins the team, perhaps , we may check this one together

**Comment by rperezb** [comment]
cc @mcastroSG

**Comment by mcastroSG** [comment]
Hi hi:

I was not able to reproduce this bug, this was tested with : 

- Client:  OSX 10.11.4,  Nightly client 
- Server: Ubuntu,  9.0.1 server

and 

- Client:  OSX 10.11.4,  2.1.1 client 
- Server: Ubuntu,  8.2.3 server

 Unable to reproduce issue with both combinations.

**Steps to reproduce**

1. put server in maintenance mode: sudo -u www-data php occ maintenance:mode --on
2. made some changes at client folder to include some jpg 
3. wait 10 minutes
4. bring server out of maintenance mode : sudo -u www-data php occ maintenance:mode --off

Expected behaviour according to issue should be not sync with server, but files included at client side are sync.

**Comment by guruz** [comment]
:+1:

**Comment by tessus** [comment]
:+1: 
thanks
